libs/mark-flow-ui/src/config/types.ts

```diff
@@ -11,16 +11,44 @@ import {
   VotesDict,
   YesNoContest,
 } from '@votingworks/types';
+import { ButtonPressEvent } from '@votingworks/ui';
 import {
   ContestsWithMsEitherNeither,
   MsEitherNeitherContest,
 } from '../utils/ms_either_neither_contests';
 
+export enum VoteUpdateInteractionMethod {
+  Touch = 'touch',
+  AssistiveTechnologyDevice = 'assistive_technology_device',
+  Mouse = 'mouse',
+  Unknown = 'unknown',
+}
+
+export function getInteractionMethod(
+  event: ButtonPressEvent
+): VoteUpdateInteractionMethod {
+  switch (event.detail) {
+    case 0:
+      /* istanbul ignore next - react-testing-library/userEvent pointer API was added in 14.0.0; we are on 13.x.x */
+      if ((event as React.TouchEvent<HTMLButtonElement>).touches) {
+        return VoteUpdateInteractionMethod.Touch;
+      }
+      return VoteUpdateInteractionMethod.AssistiveTechnologyDevice;
+    case 1:
+      return VoteUpdateInteractionMethod.Mouse;
+    default:
+      /* istanbul ignore next */
+      return VoteUpdateInteractionMethod.Unknown;
```

Comment:
My read on `detail` is that it counts the number of interactions. So if someone double clicked you'd get `detail: 2` (see https://developer.mozilla.org/en-US/docs/Web/API/UIEvent/detail). So it might be safe to just say `Mouse` in all cases over 0 if that makes life simpler.

Reply:
[ace489f](https://github.com/votingworks/vxsuite/pull/4284/commits/ace489f9e174e71a7f1eb75a4608183cef28e7ed)
thanks!